src/tests/build.proj

```diff
@@ -467,6 +472,59 @@
     <Message Importance="High" Text="App: $(FinalPath)" />
   </Target>
 
+
+  <!-- This target builds a Native AOT iOS App -->
+  <Target Name="BuildNativeAOTiOSApp" DependsOnTargets="SetupOSSpecificProps">
+    <PropertyGroup>
+      <CategoryPath>$([System.IO.Path]::GetDirectoryName($([System.IO.Path]::GetDirectoryName($(_CMDDIR)))))</CategoryPath>
+      <CategoryName>$([System.IO.Path]::GetFileName($(CategoryPath)))</CategoryName>
+      <TestRelativePath>$(CategoryName)/$([System.String]::Copy('$(_CMDDIR)').Replace("$(CategoryPath)/",""))</TestRelativePath>
+      <TestName>$([System.IO.Path]::GetFileName($(_CMDDIR)))</TestName>
+      <AppName>$([System.String]::Copy('$(TestRelativePath)').Replace('/','_'))</AppName>
+      <AppleBuildDir>$(IntermediateOutputPath)\iOSApps\$(AppName)</AppleBuildDir>
+      <FinalPath>$(XUnitTestBinBase)$(TestRelativePath)\$(AppName).app</FinalPath>
+    </PropertyGroup>
+
+    <PropertyGroup>
+      <NativeDependencies>$(IntermediateOutputPath)\..\$(TestRelativePath)\$(TestName)\native\$(TestName).o</NativeDependencies>
+      <DevTeamProvisioning>-</DevTeamProvisioning>
+      <AppleGenerateAppBundle>true</AppleGenerateAppBundle>
+      <GenerateXcodeProject>true</GenerateXcodeProject>
+      <AppleBundleDir>$(AppDir)</AppleBundleDir>
+      <RunAOTCompilation>false</RunAOTCompilation>
+      <MonoForceInterpreter>false</MonoForceInterpreter>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <_LinkerFlagsToDrop Include="@(NativeFramework->'-framework %(Identity)')" />
+      <LinkerArg Remove="@(_LinkerFlagsToDrop)" />
+      <ExtraAppLinkerArgs Include="@(LinkerArg)" />
+      <GlobalizationkNativeLibs Include="$(MicrosoftNetCoreAppRuntimePackDir)runtimes/$(TargetOS)-$(TargetArchitecture)/native/icudt.dat" />
```

Comment:
Why is this needed? I don't think we are doing this explicitly for mono. The file should be picked up by the build.
Additional nit: `GlobalizationkNativeLibs` is misspelled.

Reply:
I think Mono and Native AOT use different publish paths. During the Mono publish the `icudt.dat` file ends up in the publish directory. After the Native AOT publish, it has to be added explicitly to the publish directory to be picked up by the AppleAppBuilder.

[InnerManagedTestBuild.3.binlog.zip](https://github.com/dotnet/runtime/files/12038283/InnerManagedTestBuild.3.binlog.zip)

Reply:
I see, if I am not mistaken that should be handled by the common `ComputeResolvedFilesToPublishList` target, so if we need to explicitly include that file, it would probably be better to do something like:
```xml
<ResolvedFileToPublish Include="$(MicrosoftNetCoreAppRuntimePackDir)runtimes/$(TargetOS)-$(TargetArchitecture)/native/icudt.dat">
```
This way copying to the publish folder would be done by the build (so the manual Copy bellow will not be needed).

The right place to include the file explicitly is probably somewhere in `AppleBuild.targets` as we would want to have the same behaviour for the HelloiOS sample as well.

Reply:
The native libs get populated in `ResolveRuntimePackAssets` and copied to the publish directory in `_CopyFilesMarkedCopyLocal`.

During the build, the `ResolveRuntimePackAssets` is skipped due to the condition below.
```
<Target Name="ResolveRuntimePackAssets" DependsOnTargets="ResolveFrameworkReferences"
          Condition="'@(RuntimePack)' != ''">
```
I assume that the RuntimePack is null since the CI builds don't use the runtime packs. If `icudt.dat` is explicitly referenced in `ResolvedFileToPublish` item, I think it happens after the `_CopyFilesMarkedCopyLocal` target and the file doesn't end up in the publish directory.

Reply:
Thanks for checking that out and my apologies as I was not aware separate builds are used in this case, so setting up what I suggested would not work.
For mono we also collect all the native dependencies: https://github.com/dotnet/runtime/blob/4d5d2dc966ee99aa2f16a7bf6e7681446586c029/src/tests/build.proj#L319 and then recreate and copy them over into what supposed to be the new publish folder.